wsclient/src/main/java/org/hpccsystems/ws/client/HPCCWsDFUXRefClient.java

```diff
@@ -0,0 +1,557 @@
+import org.hpccsystems.ws.client.gen.axis2.wsdfuxref.v1_01.DFUXRefFoundFilesQueryRequest;
+import org.hpccsystems.ws.client.gen.axis2.wsdfuxref.v1_01.DFUXRefFoundFilesQueryResponse;
+import org.hpccsystems.ws.client.gen.axis2.wsdfuxref.v1_01.DFUXRefListRequest;
+import org.hpccsystems.ws.client.gen.axis2.wsdfuxref.v1_01.DFUXRefListResponse;
+import org.hpccsystems.ws.client.gen.axis2.wsdfuxref.v1_01.DFUXRefLostFilesQueryRequest;
+import org.hpccsystems.ws.client.gen.axis2.wsdfuxref.v1_01.DFUXRefLostFilesQueryResponse;
+import org.hpccsystems.ws.client.gen.axis2.wsdfuxref.v1_01.DFUXRefMessagesQueryRequest;
+import org.hpccsystems.ws.client.gen.axis2.wsdfuxref.v1_01.DFUXRefMessagesQueryResponse;
+import org.hpccsystems.ws.client.gen.axis2.wsdfuxref.v1_01.DFUXRefOrphanFilesQueryRequest;
+import org.hpccsystems.ws.client.gen.axis2.wsdfuxref.v1_01.DFUXRefOrphanFilesQueryResponse;
+import org.hpccsystems.ws.client.gen.axis2.wsdfuxref.v1_01.DFUXRefUnusedFilesRequest;
+import org.hpccsystems.ws.client.gen.axis2.wsdfuxref.v1_01.DFUXRefUnusedFilesResponse;
+import org.hpccsystems.ws.client.gen.axis2.wsdfuxref.v1_01.EspStringArray;
+import org.hpccsystems.ws.client.gen.axis2.wsdfuxref.v1_01.WsDFUXRef;
+import org.hpccsystems.ws.client.gen.axis2.wsdfuxref.v1_01.WsDFUXRefPingRequest;
+import org.hpccsystems.ws.client.gen.axis2.wsdfuxref.v1_01.WsDFUXRefStub;
+import org.hpccsystems.ws.client.platform.Version;
+import org.hpccsystems.ws.client.utils.Connection;
+import org.hpccsystems.ws.client.wrappers.ArrayOfEspExceptionWrapper;
+import org.hpccsystems.ws.client.wrappers.EspSoapFaultWrapper;
+import org.hpccsystems.ws.client.wrappers.gen.wsdfuxref.DFUXRefArrayActionRequestWrapper;
+import org.hpccsystems.ws.client.wrappers.gen.wsdfuxref.DFUXRefCleanDirectoriesResponseWrapper;
+import org.hpccsystems.ws.client.wrappers.gen.wsdfuxref.DFUXRefFoundFilesQueryResponseWrapper;
+import org.hpccsystems.ws.client.wrappers.gen.wsdfuxref.DFUXRefMessagesQueryResponseWrapper;
+import org.hpccsystems.ws.client.wrappers.gen.wsdfuxref.DFUXRefUnusedFilesResponseWrapper;
+import org.hpccsystems.ws.client.wrappers.wsdfu.WsDFUClientStubWrapper;
+
+/**
+ * @author PastraRX
+ *
+ */
+public class HPCCWsDFUXRefClient extends BaseHPCCWsClient
+{
+    private static final Logger    log                = LogManager.getLogger(HPCCWsDFUXRefClient.class);
+
+    public static final String     WSDFUXREFURI           = "/WsDFUXRef/";
+    private static int             DEFAULTSERVICEPORT = -1;
+    private static String          WSDLURL            = null;
+
+    /**
+     * Load WSDLURL.
+     */
+    private static void loadWSDLURL()
+    {
+        try
+        {
+            WSDLURL = getServiceWSDLURL(new WsDFUXRefStub());
+            DEFAULTSERVICEPORT = (new URL(WSDLURL)).getPort();
+        }
+        catch (AxisFault | MalformedURLException e)
+        {
+            log.error("Unable to establish original WSDL URL");
```

Comment:
Do we need to put the client in an "invalid" state here and check for it below, or throw an exception here?

Reply:
I think this only happens if the URL in the WSDL is invalid, and wouldn't necessarily invalidate running instances of the service, I think we're ok as is.